src/Training.cpp

```diff
@@ -342,18 +341,73 @@ void Training::process_game(GameState& state, size_t& train_pos, int who_won,
         step.to_move = to_move;
         step.planes = get_planes(&state);
 
-        step.probabilities.resize(POTENTIAL_MOVES);
-        step.probabilities[move_idx] = 1.0f;
+        if (policies.size() == 0) {
+            step.probabilities[move_idx] = 1.0f;
+        } else {
+            step.probabilities = std::move(policies[counter]);
+        }
 
         train_pos++;
         m_data.emplace_back(step);
 
         counter++;
-    } while (state.forward_move() && counter < tree_moves.size());
+    } while (state.forward_move() && counter < tree_moves.size() && counter < policies.size());
```

Comment:
Does this not break the existing code, as you're passing an empty array so it will exit immediately?

Reply:
I somehow confused myself ... I really want `||`, not `&&` here. Will fix and test.